mllib/src/main/scala/org/apache/spark/mllib/linalg/distributed/BlockMatrix.scala

```diff
@@ -332,18 +336,18 @@ class BlockMatrix @Since("1.3.0") (
     if (rowsPerBlock == other.rowsPerBlock && colsPerBlock == other.colsPerBlock) {
       val addedBlocks = blocks.cogroup(other.blocks, createPartitioner())
         .map { case ((blockRowIndex, blockColIndex), (a, b)) =>
-          if (a.size > 1 || b.size > 1) {
-            throw new SparkException("There are multiple MatrixBlocks with indices: " +
-              s"($blockRowIndex, $blockColIndex). Please remove them.")
-          }
-          if (a.isEmpty) {
-            new MatrixBlock((blockRowIndex, blockColIndex), b.head)
-          } else if (b.isEmpty) {
-            new MatrixBlock((blockRowIndex, blockColIndex), a.head)
-          } else {
-            val result = a.head.toBreeze + b.head.toBreeze
-            new MatrixBlock((blockRowIndex, blockColIndex), Matrices.fromBreeze(result))
-          }
+        if (a.size > 1 || b.size > 1) {
```

Comment:
You shifted left by 2 spaces.  If you develop using IntelliJ Idea and run ```build/sbt gen-idea``` before opening the project, then it should (mostly) do these indents automatically & correctly.